codex/utils.nim

```diff
@@ -35,6 +36,24 @@ proc orElse*[A](a, b: Option[A]): Option[A] =
   else:
     b
 
+template findIt*(s, pred: untyped): untyped =
```

Comment:
Not sure if this is needed, there is a `binarySearch` helper in `algorithm.nim` that does the same I believe.

```nim
    assert binarySearch(["a", "b", "c", "d"], "d", system.cmp[string]) == 3
    assert binarySearch(["a", "b", "c", "d"], "c", system.cmp[string]) == 2
```

Reply:
Binary search requires collections to be sorted. We could use [a simple `find`](https://nim-lang.org/docs/system.html#find%2CT%2CS), but that doesn't work either: if you look closely, [this is the place where I'm using `findIt`](https://github.com/codex-storage/nim-codex/pull/714/files#diff-dfaae75480f3156603a1a6c6bd492f6eaae100ce1fe4cc40aed4d6815e2db5e0R403), and its precisely replacing a `find` because we need to match against an attribute of the want list entry (the address), and not the entire want list entry. 

I therefore thought of putting out a `findIt`in the same spirit of [`filterIt`](https://nim-lang.org/docs/sequtils.html#filterIt.t%2Cuntyped%2Cuntyped), which seems to be the shortest path to a solution. Btw I'd argue `findIt` should be a part of sequtils, not sure why it's not there.

Reply:
OK, thats a good point, but this utils are getting out of hands, and we need to figure out a way of cleaning them up and organizing better. For now, it's probably ok.

Reply:
Yeah I feel you. Ideally we want to isolate those utilities into packages which slowly get either pushed into existing libraries or new ones, otherwise it starts to get very hard to manage.